internal/infrastructure/kubernetes/proxy/resource.go

```diff
@@ -132,6 +132,7 @@ func expectedProxyContainers(infra *ir.ProxyInfra,
 			TrustedCA:   filepath.Join("/sds", common.SdsCAFilename),
 		},
 		MaxHeapSizeBytes: maxHeapSizeBytes,
+		XdsServerHost:    ptr.To(fmt.Sprintf("%s.%s.svc.cluster.local", infra.Config.Name, infra.Config.Namespace)),
```

Comment:
the infra handle is for the proxy/data plane, and we want to specify the control plane namespace
it may be easier to populate the namespace and dns domain from the runner config and set it in here
https://github.com/envoyproxy/gateway/blob/6f5ae8ebc8ca3c30257a74d5dedc868c96e15038/internal/infrastructure/host/proxy_infra.go#L48